CmakeModules/CodeCoverage.cmake

```diff
@@ -1,238 +1,72 @@
-
-    set(options NONE)
-    set(oneValueArgs NAME)
-    set(multiValueArgs EXECUTABLE EXECUTABLE_ARGS DEPENDENCIES)
-    cmake_parse_arguments(Coverage "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
-
-    if(NOT SIMPLE_PYTHON_EXECUTABLE)
-        message(FATAL_ERROR "python not found! Aborting...")
-    endif() # NOT SIMPLE_PYTHON_EXECUTABLE
-
-    if(NOT GCOVR_PATH)
-        message(FATAL_ERROR "gcovr not found! Aborting...")
-    endif() # NOT GCOVR_PATH
-
-    # Combine excludes to several -e arguments
-    set(COBERTURA_EXCLUDES "")
-    foreach(EXCLUDE ${COVERAGE_EXCLUDES})
-        set(COBERTURA_EXCLUDES "-e ${EXCLUDE} ${COBERTURA_EXCLUDES}")
-    endforeach()
-
-    add_custom_target(${Coverage_NAME}
-
-        # Run tests
-        ${Coverage_EXECUTABLE}
-
-        # Running gcovr
-        COMMAND ${GCOVR_PATH} -x -r ${CMAKE_SOURCE_DIR} ${COBERTURA_EXCLUDES}
-            -o ${Coverage_NAME}.xml
+        COMMAND ${LCOV_EXECUTABLE} --capture --directory .
+            --output-file ${coverageTarget}.base --initial
+            ${exclusion_args}
+
+        # Invoke test executable
+        COMMAND "$<TARGET_FILE:${executable}>"
+
+        # Capture lcov counters
+        COMMAND ${LCOV_EXECUTABLE} --capture --directory .
+            --output-file ${coverageTarget}.info ${exclusionArgs}
+        # Add baseline counters
+        COMMAND ${LCOV_EXECUTABLE} --add-tracefile ${coverageTarget}.base
+            --add-tracefile ${coverageTarget}.info
+            --output-file ${coverageTarget}.total ${exclusionArgs}
+        # Generate report
+        COMMAND ${GENHTML_EXECUTABLE} --output-directory ${coverageTarget}
+            ${PROJECT_BINARY_DIR}/${coverageTarget}.total
+        # Cleanup lcov outputs
+        COMMAND ${CMAKE_COMMAND} -E remove ${coverageTarget}.base
+            ${coverageTarget}.info ${coverageTarget}.total
+
+        # Notify user of output location
+        COMMAND ${CMAKE_COMMAND} -E echo "Open ./${coverageTarget}/index.html in your browser to view the coverage report."
         WORKING_DIRECTORY ${PROJECT_BINARY_DIR}
-        DEPENDS ${Coverage_DEPENDENCIES}
-        COMMENT "Running gcovr to produce Cobertura code coverage report."
+        DEPENDS ${executable}
+        COMMENT "Processing code coverage counters and generating report."
     )
-
-    # Show info where to find the report
-    add_custom_command(TARGET ${Coverage_NAME} POST_BUILD
-        COMMAND ;
-        COMMENT "Cobertura code coverage report saved in ${Coverage_NAME}.xml."
-    )
-
-endfunction() # SETUP_TARGET_FOR_COVERAGE_COBERTURA
-
-function(APPEND_COVERAGE_COMPILER_FLAGS)
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${COVERAGE_COMPILER_FLAGS}" PARENT_SCOPE)
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${COVERAGE_COMPILER_FLAGS}" PARENT_SCOPE)
-    message(STATUS "Appending code coverage compiler flags: ${COVERAGE_COMPILER_FLAGS}")
-endfunction() # APPEND_COVERAGE_COMPILER_FLAGS
+endfunction()
```

Comment:
I trust you've tried this in another project or so and it works similarly to the original one (which I had little exposure to, since it was mainly copy-pasted). :smiley_cat:

Reply:
I've run it locally, and it does exactly what you used the original module for. All I did was remove all bits we don't use, and aggressively simplify to better match our use case.